File: JumpGame.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool canJump(vector<int>& nums) 
    {
        int limit=nums[0];
        for(int i=1;i<nums.size()-1;i++)
        {
            if(i<=limit)
            {
                limit=max(limit,i+nums[i]);
            }
        }    
        return limit>=nums.size()-1;
    }
};

int main()
{
    
    return 0;
}

File: JumpGame.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let limit = nums[0];
    for (let i = 1; i < nums.length - 1; i++) {
        if (i <= limit) {
            limit = Math.max(limit, i + nums[i]);
        }
    }
    return limit >= nums.length - 1;
};
